app/code/Magento/Customer/Model/Plugin/CustomerAuthorization.php

```diff
@@ -53,9 +72,14 @@ public function aroundIsAllowed(
             && $this->userContext->getUserId()
             && $this->userContext->getUserType() === UserContextInterface::USER_TYPE_CUSTOMER
         ) {
-            return true;
-        } else {
-            return $proceed($resource, $privilege);
+            $customer = $this->customerFactory->create()->load($this->userContext->getUserId());
```

Comment:
The `customerFactory` will return the customer ORM model. Considering this fact, we are calling the deprecated `load` method in order to retrieve the customer. 

I would suggest avoiding this approach and use the corresponding resource model for loading the entity instead

Reply:
@rogyar updated